fhir-persistence/src/test/java/com/ibm/fhir/persistence/search/test/AbstractSearchQuantityTest.java

```diff
@@ -82,80 +75,110 @@ public void testSearchQuantity_Quantity_revinclude() throws Exception {
+        assertSearchReturnsSavedResource("Quantity", "ap24.4999||s");
+        assertSearchReturnsSavedResource("Quantity", "ap24.5||s");
         assertSearchReturnsSavedResource("Quantity", "ap25||s");
-//        assertSearchReturnsSavedResource("Quantity", "ap25.4999||s");
-//        assertSearchReturnsSavedResource("Quantity", "ap25.5||s");
-//        assertSearchReturnsSavedResource("Quantity", "ap26|http://unitsofmeasure.org|s");
-        
+        assertSearchReturnsSavedResource("Quantity", "ap25.4999||s");
+        assertSearchReturnsSavedResource("Quantity", "ap25.5||s");
+        assertSearchReturnsSavedResource("Quantity", "ap26|http://unitsofmeasure.org|s");
+        assertSearchDoesntReturnSavedResource("Quantity", "ap30|http://unitsofmeasure.org|s");
+    }
+
+    @Test
+    public void testSearchQuantity_Quantity_withPrefix_LT() throws Exception {
         assertSearchDoesntReturnSavedResource("Quantity", "lt24|http://unitsofmeasure.org|s");
         assertSearchDoesntReturnSavedResource("Quantity", "lt24.4999||s");
         assertSearchDoesntReturnSavedResource("Quantity", "lt24.5||s");
         assertSearchDoesntReturnSavedResource("Quantity", "lt25||s");
         assertSearchReturnsSavedResource("Quantity", "lt25.4999||s");
         assertSearchReturnsSavedResource("Quantity", "lt25.5||s");
         assertSearchReturnsSavedResource("Quantity", "lt26|http://unitsofmeasure.org|s");
-        
+
+    }
+
+    @Test
+    public void testSearchQuantity_Quantity_withPrefix_GT() throws Exception {
         assertSearchReturnsSavedResource("Quantity", "gt24|http://unitsofmeasure.org|s");
         assertSearchReturnsSavedResource("Quantity", "gt24.4999||s");
         assertSearchReturnsSavedResource("Quantity", "gt24.5||s");
-        assertSearchDoesntReturnSavedResource("Quantity", "gt25||s");
+
+        // For the following test, as the upper bound is now the lower bound of the range
+        // the following must use a lower value of the range/precision 
+        // To keep from hitting, we have to make it more precise. 
+        assertSearchDoesntReturnSavedResource("Quantity", "gt25.05||s");
         assertSearchDoesntReturnSavedResource("Quantity", "gt25.4999||s");
         assertSearchDoesntReturnSavedResource("Quantity", "gt25.5||s");
         assertSearchDoesntReturnSavedResource("Quantity", "gt26|http://unitsofmeasure.org|s");
-        
+
+    }
+
+    @Test
+    public void testSearchQuantity_Quantity_withPrefix_LE() throws Exception {
         assertSearchDoesntReturnSavedResource("Quantity", "le24|http://unitsofmeasure.org|s");
         assertSearchDoesntReturnSavedResource("Quantity", "le24.4999||s");
         assertSearchDoesntReturnSavedResource("Quantity", "le24.5||s");
-        assertSearchReturnsSavedResource("Quantity", "le25||s");
+
+        // We have to make the following test more precise since the implied range is used. 
+        assertSearchReturnsSavedResource("Quantity", "le25.01||s");
```

Comment:
i don't get it. shouldn't the presence of gt prefix make us treat le25 as if it were le25.0000000000... ?